state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1310,6 +1326,50 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(14)]
+		#[pallet::weight(T::WeightInfo::open_private_channel())]
+		pub fn open_private_channel(origin: OriginFor<T>) -> DispatchResult {
+			let broker_id = T::AccountRoleRegistry::ensure_broker(origin)?;
+
+			ensure!(
+				!BrokerPrivateChannels::<T, I>::contains_key(&broker_id),
+				Error::<T, I>::PrivateChannelExistsForBroker
+			);
+
+			// TODO: burn fee for opening a channel?
+			let next_channel_id =
+				ChannelIdCounter::<T, I>::try_mutate::<_, Error<T, I>, _>(|id| {
+					*id = id.checked_add(1).ok_or(Error::<T, I>::ChannelIdsExhausted)?;
+					Ok(*id)
+				})?;
```

Comment:
we should make this a function, is duplicate

Reply:
Deduplicated